backend/linkbox/linkbox.go

```diff
@@ -0,0 +1,946 @@
+	if err != nil {
+		return err
+	}
+
+	if len(entries) != 0 && check {
+		return fs.ErrorDirectoryNotEmpty
+	}
+
+	opts := &rest.Opts{
+		Method:  "GET",
+		RootURL: linkboxAPIURL,
+		Path:    "folder_del",
+		Parameters: url.Values{
+			"token":  {f.opt.Token},
+			"dirIds": {strconv.Itoa(dirIds)},
+		},
+	}
+
+	response := getResponse{}
+	err = getUnmarshaledResponse(ctx, f, opts, &response)
+
+	if err != nil {
+		return fmt.Errorf("err in response")
+	}
+
+	if response.Status != 1 {
+		// it can be some different error, but Linkbox
+		// returns very few statuses
+		return fs.ErrorDirExists
+	}
+	return nil
+}
+
+// Rmdir removes the directory (container, bucket) if empty
+//
+// Return an error if it doesn't exist or isn't empty
+func (f *Fs) Rmdir(ctx context.Context, dir string) error {
+	fs.Infof(f, "rmdir - {%s}", dir)
+	return f.purgeCheck(ctx, dir, true)
+}
+
+func (o *Object) SetModTime(ctx context.Context, modTime time.Time) error {
+	return fs.ErrorCantSetModTime
+}
+
+// Open opens the file for read.  Call Close() on the returned io.ReadCloser
+func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (io.ReadCloser, error) {
+	fs.Infof(o, "Open - {%s}", o.Remote())
+
+	var res *http.Response
+	downloadURL := o.fullURL
+	if downloadURL == "" {
+		_, name := splitDirAndName(o.Remote())
+		newObject, err := getObject(ctx, o.fs, name, o.pid, o.fs.opt.Token)
+		if err != nil {
+			return nil, err
+		}
+		if newObject == (Entity{}) {
+			fs.Debugf(o.fs, "Open entity is empty: name = {%s}", name)
+			return nil, fs.ErrorObjectNotFound
+		}
+
+		downloadURL = newObject.URL
+	}
+
+	opts := &rest.Opts{
+		Method:  "GET",
+		RootURL: downloadURL,
+		Options: options,
+	}
+
+	err := o.fs.pacer.Call(func() (bool, error) {
+		var err error
+		res, err = o.fs.srv.Call(ctx, opts)
+		return o.fs.shouldRetry(ctx, err)
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("Open failed: %w", err)
+	}
+
+	return res.Body, nil
+}
+
+// Update in to the object with the modTime given of the given size
+//
+// When called from outside an Fs by rclone, src.Size() will always be >= 0.
+// But for unknown-sized objects (indicated by src.Size() == -1), Upload should either
+// return an error or update the object properly (rather than e.g. calling panic).
+func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) error {
+	fs.Infof(o, "Update: Remote = {%s}", o.Remote())
+
+	if src.Size() == 0 {
+		return fs.ErrorCantUploadEmptyFiles
+	}
+
+	remote := o.Remote()
+	tmpObject, err := o.fs.NewObject(ctx, remote)
+
+	if err == nil {
+		fs.Debugf(o.fs, "Update: removing old file")
+		_ = tmpObject.Remove(ctx)
+	}
+
+	h := md5.New()
+	first10m := io.LimitReader(in, 10_485_760)
+	first10mBytes, err := io.ReadAll(first10m)
+	if err != nil {
+		fs.Debugf(o.fs, "Update err in reading file: %v", err)
+		return err
+	}
+
+	if _, err := io.Copy(h, bytes.NewReader(first10mBytes)); err != nil {
```

Comment:
You can just use `md5.Sum(first10mBytes)`

Reply:
Done. Thanks